openedx/core/djangoapps/programs/utils.py

```diff
@@ -142,6 +179,11 @@ def progress(self, programs=None, count_only=True):
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(utc)
+        enrolled_run_modes = defaultdict(set)
+        for enrollment in self.enrollments:
+            enrolled_run_modes[str(enrollment.course_id)].add(enrollment.mode)
```

Comment:
`self.course_run_ids` are simply the course run IDs corresponding to the user's enrollments. Use it instead of looping over the enrollments and converting the course keys to strings again.